cadc-rest/src/main/java/ca/nrc/cadc/rest/RestServlet.java

```diff
@@ -160,22 +161,30 @@ public void init(ServletConfig config) throws ServletException {
         }
         
         try {
-            Class<InitAction> initAction = loadAction(config, "init");
-            if (initAction != null) {
-                InitAction action = initAction.newInstance();
-                action.setServletContext(getServletContext());
-                action.setServletContext(getServletContext());
-                action.setAppName(appName);
-                action.setComponentID(componentID);
-                action.setInitParams(initParams);
-                action.doInit();
+            Class<InitAction> initActionClass = loadAction(config, "init");
+            if (initActionClass != null) {
+                initAction = initActionClass.getDeclaredConstructor().newInstance();
+                initAction.setServletContext(getServletContext());
+                initAction.setAppName(appName);
+                initAction.setComponentID(componentID);
+                initAction.setInitParams(initParams);
+                initAction.doInit();
             }
         } catch (Exception ex) {
             log.error("init failed", ex);
             throw new ServletException("init failed", ex);
         }
     }
 
+    @Override
+    public void destroy() {
+        try {
+            initAction.doShutdown();
```

Comment:
`if (initAction != null)` ??

Reply:
Yes checking the initAction isn't null would stop a large unnecessary log message. Fixed.